build-setup/windows.md

```diff
@@ -21,21 +21,32 @@ You will also need to add several directories to your `PATH` environment variabl
 - `/c/PROGRA~2/MICROS~2/2019/Professional/TEAMTO~1/PERFOR~1/x64`
 - `/c/PROGRA~2/MICROS~2/2019/Professional/TEAMTO~1/PERFOR~1`
 - `/c/PROGRA~1/Java/jdk1.8.0_121/bin`
+- `/c/PROGRA~1/CMake/bin/cmake`
 - `/c/usr/local/bin`
 - `/c/usr/local/lib`
 
-Finally, you need to create an alias `make` for `gnumake`.
-
 For example, you might create a `.bashrc` file in your home directory containing this:
 
 ```
 export CPP_SRC_HOME=$HOME/ml-cpp
 VCVER=`/bin/ls -1 /c/PROGRA~2/MICROS~2/2019/Professional/VC/Tools/MSVC | tail -1`
 VCBINDIR=/c/PROGRA~2/MICROS~2/2019/Professional/VC/Tools/MSVC/$VCVER/bin/Hostx64/x64:/c/PROGRA~2/MICROS~2/2019/Professional/Common7/IDE:/c/PROGRA~2/WI3CF2~1/8.0/bin/x64:/c/PROGRA~2/WI3CF2~1/8.0/bin/x86:/c/PROGRA~2/MICROS~2/2019/Professional/TEAMTO~1/PERFOR~1/x64:/c/PROGRA~2/MICROS~2/2019/Professional/TEAMTO~1/PERFOR~1
 export JAVA_HOME=/c/PROGRA~1/Java/jdk1.8.0_121
-export PATH="$CPP_SRC_HOME/build/distribution/platform/windows-x86_64/bin:$VCBINDIR:/mingw64/bin:$JAVA_HOME/bin:/c/usr/local/bin:/c/usr/local/lib:/bin:/c/Windows/System32:/c/Windows"
-alias make=gnumake
+export PATH="$CPP_SRC_HOME/build/distribution/platform/windows-x86_64/bin:$VCBINDIR:/mingw64/bin:$JAVA_HOME/bin:/c/usr/local/bin:/c/usr/local/lib:/bin:/c/Windows/System32:/c/Windows:/c/PROGRA~1/CMake/bin/cmake"
+```
+
+The build system uses CMake. To build (once the following dependencies have been installed) either call `cmake` directly from the top level of the source tree, e.g.
 ```
+cmake -B cmake_build -DCMAKE_TOOLCHAIN_FILE=cmake/windows-x86_64.cmake
+cmake --build cmake_build --config Release -v -j`nproc`
+```
+
+or, more simply, use Gradle
+```
+./gradlew :compile
+```
+
+Note that we configure the build to be of type `Release`, and specify the compiler flag `/Zi` in the cmake configuration files in order to obtain a fully optimized build along with debugging symbols. This is preferable to say, using the `RelWithDebInfo` configuration as `RelWithDebInfo` has some undesirable side effects (such as disabling inlining) 
```

Comment:
As for the other platforms, consolidate this in https://github.com/elastic/ml-cpp/blob/main/CONTRIBUTING.md#working-with-the-ml-cpp-codebase